src/System.Private.ServiceModel/src/System/ServiceModel/Description/ServiceReflector.cs

```diff
@@ -451,14 +451,25 @@ static internal object[] GetCustomAttributes(CustomAttributeProvider attrProvide
         {
             try
             {
-                if (typeof(Attribute).IsAssignableFrom(attrType))
+                List<object> attrTypeAttributes = new List<object>();
```

Comment:
I recommend this code to make a new List and filter the attributes should really happen only in the case attributeType is not assignable to Attribute.  In other words, uncomment the original code below, but move this List-generating code into the "else" part of the code where an empty array was being returned.  And rather than calling the modified version of GetCustomAttributes that takes "attrType", it should call the version of GetCustomAttributes that does not take an attrType.

The original "if" block commented out below should be uncommented exactly as it was.  We want the #if part of the code to be unchanged, and only modify the #else part.